app/models/payment/braintree_customer.rb

```diff
@@ -1,5 +1,9 @@
 class Payment::BraintreeCustomer < ActiveRecord::Base
   belongs_to :member
   has_many :braintree_payment_methods, class_name: Payment::BraintreePaymentMethod, primary_key: :customer_id, foreign_key: :customer_id
-  belongs_to :default_payment_method, class_name: Payment::BraintreePaymentMethod
+
+
+  def default_payment_method
+    braintree_payment_methods.order('created_at desc').first
+  end
```

Comment:
I think this is the main change. rather than directly associating the default payment method, you could instead fetch the latest payment method record directly. It means you have less housekeeping else where in the code.